transcom/mymove
# MB-9411 Switch sftp connection to use private key

## [MB-9411](https://dp3.atlassian.net/browse/MB-9411)

## Summary

This does the actual small change to make GEX connect via key based authentication. Password based authentication is left as a backup.

Unfortunately, the GEX stuff is not able to be tested in exp.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### How to test

1. Generate a key pair with `ssh-keygen -f ./test_key_pair -t ecdsa -b 521` (You may want to do this inside of the `tmp` folder since that gets gitignored)
2. Run `make docker_local_ssh_server_with_key`
3. Open the CLI to the Docker container you just made
4. Inside the CLI for the Docker container run `cat /etc/ssh/ssh_host_ecdsa_key.pub`. 
5. Inside of the project's `.envrc.local` set these environment variables, the GEX_SFTP_HOST_KEY should use the key that you got in step 6.

```
export GEX_SFTP_USER_ID=testu
export GEX_SFTP_PASSWORD=testp
export GEX_SFTP_IP_ADDRESS=localhost
export GEX_SFTP_PORT=2222
export SEND_TO_SYNCADA=true
export GEX_SFTP_HOST_KEY='this should be the value of what you retrieved from step 3'
export GEX_PRIVATE_KEY=$(<./tmp/test_key_pair)
export TEST_GEX_PUBLIC_KEY=$(<./tmp/test_key_pair.pub)
```

Perform the following steps in a separate terminal
6. Run `direnv allow` 
7. Run `make build_tools`
8. Run `milmove-tasks connect-to-gex-via-sftp`

### Backend



[MB-9411]: https://dp3.atlassian.net/browse/MB-9411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
